app/services/custom_failure.rb

```diff
@@ -0,0 +1,15 @@
+class CustomFailure < Devise::FailureApp
+  def respond
+    if http_auth?
+      http_auth
```

Comment:
I like early returns (`return http_auth if http_auth?`) instead of more `if` branches but it's a matter of preference :)